addons/web/static/src/js/fields/field_utils.js

```diff
@@ -203,24 +203,57 @@ function formatFloatFactor(value, field, options) {
  * @param {Object} [field]
  *        a description of the field (note: this parameter is ignored)
  * @param {Object} [options]
- * @param {boolean} [options.noLeadingZeroHour] if true, format like 1:30
- *        otherwise, format like 01:30
+ * @param {string} [unit] gives the unit of the original float in hours, minutes or seconds.
+ *        The default is 'hours', but doesn't really have to be set.
+ *         Note: If the float must convert to xx:yy, no need to specify if those are minutes or hours (cfr. param below).
+ * @param {boolean} [threeDigits] specifies if the output format should presents the three times info => hh:mm:ss
+ *         If false (default), the float will present like xx:yy
+ *
+ *          1,52 h => 01:30:12      {'threeDigits':true}
+ *                 => 01:30         {}
+ *          1,52 m => 00:01:30      {'unit':'minutes', 'threeDigits':true}
+ *                 =>    01:30      {} /!\ be carefull : If yous et the unit to 'minutes', it will automatically
+                                        convert to hours. Don't do it unless you want the three digits.
+                                        This si mandatory to not break the behavior already in place.
+ *        128,52 s => 00:02:09      {'unit':'seconds', 'threeDigits':true}
+ *                 =>    02:09      {'unit':'seconds'}
+ *
+ * @param {boolean} [leadingZero] if true, format like 01:30
+ *        otherwise, format like 1:30
  * @returns {string}
  */
-function formatFloatTime(value, field, options) {
-    options = options || {};
-    var pattern = options.noLeadingZeroHour ? '%1d:%02d' : '%02d:%02d';
+
+function formatFloatTime(value, field, {unit = '', threeDigits = false, leadingZero = true} = {}) {
```

Comment:
Hello there @dbeguin  !
I am not the most familiar with this. But @tde-banana-odoo asked to take advantage we change that to update to the ES6 JS. I believe it's what what required in order to do so. If not, I'm willing to know what was wrong, I'm learning :)

You mean replace the boolean with a string ?

So it took me a long time to find that duration "widget". Actually, it's not a widget. It's something in the qweb, please see the "value_to_html" method. I really don't think I can use any of this for  the duration widget.

Naming to "dimensions" is just a matter of opinion I guess ? Plus, I think I'd rather have a boolean rather than an integer here, as there is only 2 solutions possibles. It would avoid a random guy putting a random number without understanding why it "doesn't work".

Your comment may have given me an idea tho. The shortcut I could think about is having a "format" option indeed that could be filled like :
hh:mm:ss
hh:mm
mm:ss
h:mm:ss
h:mm
m:ss

This option alone could answer the leadingZero + threeDigits + unit in once. That could be way easier (or clearer) to use ?

Please tell me what you think of it :)

Reply:
EDIT: la méthode que je propose ne permet pas de savoir dans quelle unité on donne le float de départ en fait, donc cette option resterait. Du coup je ne sais pas si c'est toujours aussi intéressant "juste" pour racourcir le leadingZero et threeDigits. C'est peut-etre tout de meme plus clair à utiliser pour ces options.

Reply:
> Naming to "dimensions" is just a matter of opinion I guess ? Plus, I think I'd rather have a boolean rather than an integer here, as there is only 2 solutions possibles. It would avoid a random guy putting a random number without understanding why it "doesn't work".

Well the idea was more to offer more flexibility. dimensions can be 1, 2 or 3, not only 3. the idea was to allow the user to choose how many dimension he wants to display. But the issue is which dimensions to display (e.g. with dimensions = 2 -> hh:mm or mm:ss ?)

So maybe the format could be the best solution.
and you still need the 'unit' as you need to tell the method what you give, indeed.

Reply:
Ok I'll take a look to use that format option. 

hh:mm or mm:ss wouldn't matter so much, the calculation result is the same (in fact it's treated the same way in the code.
A dimension 1 would be useless to convert the float..?

The "unit" is already implemented with hours as default.

Anyway, I'll take a look for the format :)